### PR TITLE
Simplified main/cosim main calling

### PR DESCRIPTION
**Pull request by drichmond**
* moved cosim_main to header file (which seems REALLY strange)
* Moved svScope VCS code into bsg_manycore_lib with macro guards

Passes regression on 4x4_fast_n_fake (except for test_symbol_to_eva)

This makes cosim_main VCS specific, not cosim specific (because Verilator uses main)

### CODE REVIEW

File: libraries/bsg_manycore.cpp [hunk 1]
```diff
@@ -536,6 +561,7 @@ int  hb_mc_manycore_init(hb_mc_manycore_t *mc, const char *name, hb_mc_manycore_
         r = err;
         hb_mc_manycore_cleanup_fifos(mc);
         hb_mc_manycore_cleanup_mmio(mc);
+        hb_mc_manycore_cleanup_dpi(mc);
```
Comment (mrutt92) [inline libraries/bsg_manycore.cpp:564]: what happens if COSIM is not defined?

Comment (drichmond) [reply libraries/bsg_manycore.cpp:564]: Oops, missed that

Comment (drichmond) [reply libraries/bsg_manycore.cpp:565]: Fixed

File: regression/library/test_printing.c [hunk 2]
```diff
@@ -47,6 +47,15 @@
 #include "test_printing.h"
 static int test_printing(void)
 {
+        int rc;
```
Comment (mrutt92) [inline regression/library/test_printing.c:50]: why is this needed now?

File: libraries/bsg_manycore.cpp [hunk 3]
```diff
@@ -254,6 +258,21 @@ static int hb_mc_manycore_rx_fifo_drain(hb_mc_manycore_t *mc, hb_mc_fifo_rx_t ty
...
+{
+        svScope scope;
+        scope = svGetScopeFromName("tb");
+        svSetScope(scope);
+        return HB_MC_SUCCESS;
+}
+
+static void hb_mc_manycore_cleanup_dpi(hb_mc_manycore_t *mc)
+{
+        return;
```
Comment (vb000) [inline libraries/bsg_manycore.cpp:272]: What's the job of the function? Are we going to do something in this later?

Comment (drichmond) [reply libraries/bsg_manycore.cpp:272]: Yep. For an incoming feature :p 

(All of my current PRs are for a fancy new feature)

File: regression/library/test_printing.c (hunk 2 above)
Comment (drichmond) [reply regression/library/test_printing.c]: Fixed

**Comment by drichmond** [comment]
This passes regression with 4x4_fast_n_fake, and timing_v0_8_4

@mrutt92, sufficient for you?
